contracts/governance/FeeSharingProxy.sol

```diff
@@ -92,29 +95,48 @@ contract FeeSharingProxy is SafeMath96, IFeeSharingProxy {
 	/// @notice An event emitted when user fee get withdrawn.
 	event UserFeeWithdrawn(address indexed sender, address indexed receiver, address indexed token, uint256 amount);
 
+	/// @notice An event emitted when user set wrbtc address.
+	event SetWrbtcToken(address indexed sender, address indexed oldWRBTCAddress, address indexed newWRBTCAddress);
+
 	/* Functions */
 
-	constructor(IProtocol _protocol, IStaking _staking) public {
+	constructor(IProtocol _protocol, IStaking _staking, address _wRBTCAddress) public {
 		protocol = _protocol;
 		staking = _staking;
+		wRBTCAddress = _wRBTCAddress;
+	}
+
+	/**
+	 * @notice Set new wRBTC address
+	 *
+	 * @param newWRBTCAddress new wRBTC addres
+	 * */
+	function setWRBTCAddress(address newWRBTCAddress) external onlyOwner {
+		address oldWRBTCAddress = wRBTCAddress;
+		wRBTCAddress = newWRBTCAddress;
+		emit SetWrbtcToken(msg.sender, oldWRBTCAddress, newWRBTCAddress);
```

Comment:
as a rule of thumb when setting contract addresses, we always need to check if the new address is a contract

Reply:
This one will be removed